qiskit/transpiler/passmanager.py

```diff
@@ -24,14 +25,15 @@
 from .exceptions import TranspilerError
 from .runningpassmanager import RunningPassManager, FlowController
 
-
+# TODO: why not
+#  _CircuitsT = Union[List[QuantumCircuit], QuantumCircuit]
```

Comment:
But the proposed definition does not type, as you said. I still don't get what's the thing to do that is very annotated.

```suggestion
```

Reply:
I'm not sure what do you mean? I do not see any type errors appearing when I use the proposed definition. The original definition. With current definition you can't assign `List[QuantumCircuit]` or `QuantumCircuit` to variables typed as `_CircuitsT` which was the whole point of the definition from what I can see.

As for comment, I did not mean it to be merged, just to draw your attention to the fact that this should be fixed and to decide on proper solution before merging the PR

Reply:
I see, let's go with `_CircuitsT = Union[List[QuantumCircuit], QuantumCircuit]` instead.